core/manifest/ConfigStaticManifest.php

```diff
@@ -1,4 +1,71 @@
 <?php
+
+/**
+ * Allows access to config values set on classes using private statics.
+ *
+ * @package framework
+ * @subpackage manifest
+ */
+class SS_ConfigStaticManifest_Reflection extends SS_ConfigStaticManifest {
+
+	/**
+	 * Constructs and initialises a new config static manifest, either loading the data
+	 * from the cache or re-scanning for classes.
+	 *
+	 * @param string $base The manifest base path.
+	 * @param bool   $includeTests Include the contents of "tests" directories.
+	 * @param bool   $forceRegen Force the manifest to be regenerated.
+	 * @param bool   $cache If the manifest is regenerated, cache it.
+	 */
+	public function __construct($base, $includeTests = false, $forceRegen = false, $cache = true) {
+		// Stubbed as these parameters are not needed for the newer SS_ConficStaticManifest version.
+	}
+
+	/**
+	 * Completely regenerates the manifest file.
+	 */
+	public function regenerate($cache = true) {
+		Deprecation::notice('3.4', 'This is no longer available as SS_ConfigStaticManifest now uses Reflection. For backwards compatibility define SS_USE_OLD_CONFIGSTATICMANIFEST in your _ss_environment.php file.');
```

Comment:
Deprecation notices should point at the version that the code will be removed; In this case, `4.0`.